dradis/dradis-ce
# My tasks view

### **How to Test**

### **Notifications tweaks**

**Expected behaviour:** The nav bar should be forced the state where a user has a notification, I couldn’t fully force this without a more in depth knowledge of how the states are toggled and where the notification total is stored, I’ve commented in the code what I’ve changed for test purposed.

**Steps to test:** 

1. Click the ‘Notifications’ item in the top navigation menu
2. The dropdown menu should be triggered displaying a list of your notifications. 
3. Clicking the user’s name would bring you to their page, I’m unsure if this functionality if currently in the system.
4. Clicking the task item title will take the user to the task’s respective page. 

### **Sidebar tweaks**

**Expected behaviour:** The sidebar menu should display a new item titled ‘My Tasks'.

**Steps to test:** 

The menu item should appear underneath ‘All Issues’

Clicking the new item will take the user straight to the ‘My Tasks' page.

**### Homepage widget**

**Expected behaviour:** The user should be presented with their highest priority tasks first, there is an option to change how tasks are sorted.

**Steps to test:** 

1. The widget should be displayed to the user at the top of the home page, to the right of the existing issues summary view.
2. The sorting dropdown, while not functional, should present a dropdown when clicked of each proposed sorting option ('Title', ‘Due Date’, ‘Priority’) with priority being the default.
3. Clicking the ‘View all tasks’ button should take the user to the respective My Tasks page, also created as part of this ticket. 
4. Clicking a task’s title takes the user directly to the page for that particular task, whether it be a card or otherwise. 
5. The user should also see information on the task’s priority and due date at a glance.

**Extra notes:**

- I decided to push the widget up the page priority, in previous systems I’ve worked in, such as an LMS, users are often visiting to carry out a specific task, therefore presenting their task list above the fold is important. 
- Clicking the ‘View all tasks’ link should take the user to the new My Tasks page.
- Although the default table headers are there, I removed the sorting here and opted for a separate sort, being new to the system I didn’t find it 100% clear that the table headers double as sorting options, especially when a caret arrow was not displayed by default. There’s no functionality for my new dropdown sort, but it would behave as the table titles do currently.
- The user should be able to quickly jump to the task inside its methodology by clicking the link in their list view.
- The task’s type was integrated into the title, I initially had a ‘type’ column here but it all looked a bit squished, given the limited space.
- The task item’s tags and due date are clearly colour coded and displayed to the user for quick context on what’s the top of their to-do list.


### 
**My Tasks page**

Expected behaviour: The user should be presented with a list of all their tasks grouped by methodology. The user will only be presented with methodologies containing tasks where they are an assignee.

**Steps to test:** 

1. Click either the ‘My Tasks' side menu item, or the ‘View All Tasks’ link in the new My Tasks homepage widget.
2. Column toggles should be in place, there’s also supposed to be the usual quick actions here to edit and delete a task, this would mainly be used to quickly update a task’s status without having to click into the task’s page. I mocked this up in [Figma.](https://www.figma.com/file/sZZ8Ed3CvNzrOHgS0pHgrb/Dradis?type=design&node-id=0%3A1&mode=design&t=oZvwyb6omVq8erp3-1)
3. The user should be able to toggle the displayed methodology by using the top dropdown, when clicked there should be a list of methodologies that the user has tasks inside of, the main title indicates which methodology is checked but perhaps some contextual text by default here would be better? 
4. Default columns should be ('Type', ‘Title’, ‘Tags’, ‘Due Date’, ‘Status’ and ‘Assignees’).
5. There should be a search bar to allow users to quickly find a task if they know the title. 
6. The methodology box should have the usual pagination and sorting options that the user is accustomed to elsewhere in the system.

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List
